src/material/autocomplete/autocomplete.ts

```diff
@@ -336,4 +343,37 @@ export class MatAutocomplete extends _MatAutocompleteBase {
   @ContentChildren(MatOption, {descendants: true}) options: QueryList<MatOption>;
   protected _visibleClass = 'mat-mdc-autocomplete-visible';
   protected _hiddenClass = 'mat-mdc-autocomplete-hidden';
+
+  constructor(
+    _changeDetectorRef: ChangeDetectorRef,
+    _elementRef: ElementRef<HTMLElement>,
+    @Inject(MAT_AUTOCOMPLETE_DEFAULT_OPTIONS) defaults: MatAutocompleteDefaultOptions,
+    platform?: Platform,
+  ) {
+    super(_changeDetectorRef, _elementRef, defaults, platform);
+
+    this._hideSingleSelectionIndicator = defaults.hideSingleSelectionIndicator ?? false;
+  }
+
+  /** Whether checkmark indicator for single-selection options is hidden. */
+  @Input()
+  get hideSingleSelectionIndicator(): boolean {
+    return this._hideSingleSelectionIndicator;
+  }
+  set hideSingleSelectionIndicator(value: BooleanInput) {
+    this._hideSingleSelectionIndicator = coerceBooleanProperty(value);
+    this._syncParentProperties();
```

Comment:
In what case does this become a problem? E.g. is it only if the binding changes while the autocomplete is open or does it not get picked up on open either?

Reply:
It doesn't get picked up on open. Although, it does get picked up when selecting an option, selecting the same option again, or selecting a different option. It seems to me that selecting an option triggers chance detection by conincidence, which cases the child component to detect the change to `hideSingleSelectionIndicator`.

```
// option.ts
  /** Whether to display checkmark for single-selection. */
  get hideSingleSelectionIndicator(): boolean {
    return !!(this._parent && this._parent.hideSingleSelectionIndicator);
  }
```

Both MatOption and MatAutocomplete use on-push change detection. I believe that means that mutating `_parent` will not trigger change detection on MatOption. That's why we have to manually call `markForCheck`. Am I understanding the change detection correctly :)?

Here's a video capture of the behavior I see when I remove `_syncParentProperties`.


https://user-images.githubusercontent.com/7720245/210853511-6c112c5b-36f8-4e15-ac82-ffe0307f65c1.mov

Reply:
I see. I'm surprised that this didn't come up for `disableRipple` or `multiple` who follow the same pattern.

Reply:
I can reproduce this problem with `disableRipple`. I think this has been happening all along, but there haven't been any issues filed for it because `disableRipple` and `multiple` are changed very rarely in practice.

I added a `disableRipple` option to the autocomplete demo and here is a video of the reproduction. It's kinda hard to see in the video, so here are the steps that happen.

(Ripple starts off enabled.
1. Click "Alabama"
2. (displays ripple animation)
3.  Click "Alabama" again
4. (displays ripple animation)
5. Click "TOGGLE DISABLE RIPPLE"
6. Click "Alabama"
7.  (displays ripple animation. Is not supposed to display ripple animation)
8. Click "Alabama" again
9.  (does not display ripple animation)


https://user-images.githubusercontent.com/7720245/210895519-67462c69-2abb-4a5a-af5f-c0e5216b0d74.mov